proposed/http-middleware/middleware.md

```diff
@@ -109,6 +108,3 @@ interface DelegateInterface
     public function process(ServerRequestInterface $request);
```

Comment:
I think this method name should be renamed right? Else a class cannot implement both interfaces.

`__invoke()` was mentioned on Slack but then it is contradicted by a section of the document (which is being removed in this PR):

> #### Why doesn't delegate use `__invoke`?
>
> Classes that define `__invoke` can be more loosely type hinted as `callable`, which results in less strict typing. This is generally undesirable, especially when the method uses strict typing.